gcache/src/gcache_rb_store.cpp

```diff
@@ -704,14 +786,116 @@ namespace gcache
            offset = -1;
         }
 
+        if (enc_encrypted != encrypt_) {
+            // if we are switching enc <-> not enc, no point in recovering
+            force_reset = true;
+        }
+
+        if (encrypt_) {
+            /* Q: Why we store master key in gcache.preamble instead of
+               doing the whole key management (e.g rotation) on server side
+               and only informing Galera about the new key?
+               A: We need access to the master key when Galera is initialized
+               to be able to perform GCache recovery. At this stage storage
+               engines are not initialized yet (this is yet another question, why
+               the initialization sequence is as such).
+               If we did key management on server side, we would need to store
+               key name info somewhere. Wsrep_schema and dedicated table seems to
+               be ideal place, but again, we don't have access to it when Galera
+               starts, so another solution is keeping it in dedicated file.
+               But having yet another file on server side if we have this preamble
+               does not seem to be good idea. */
```

Comment:
Isn't the preamble part of the actual gcache? It's in the same memory block, only separated by an offset.

Reply:
Yes, RingBuffer file already starts with the preamble.
I'm not sure I understand the question. Could you elaborate?

Reply:
Sorry, I forgot to edit this question after reading further, the issue here is with the comment, not the code: the master key isn't stored in the preamble, only its name.

Reply:
comment ammended